### PR TITLE
feat: [PL-39190]: Helm chart url for smp

### PR DESCRIPTION
**Pull request by AnupamIO**
For onprem helm chart should be referencing to customer-manager-url not app.harness.io. This is because helm chart is stored in delegate proxy and customer will fetch from there.

For saas it should keep on referring to app.harness.io

Testing:
Tested in local using postman.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51169)

### CODE REVIEW

**Review by mdmilic — COMMENTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51169)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java` line 39 at r1](https://reviewable.io/reviews/harness/harness-core/51169#-NalM_x16LlZ8VCV9FnP:-NalM_x16LlZ8VCV9FnQ:b-hbbzcn) ([raw file](https://github.com/harness/harness-core/blob/2f3a01eba252187b60a8dc7d3c7a25dffda37e32/400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java#L39)):*
> ```Java
>   private static final String TERRAFORM_TEMPLATE_FLE = "/delegatetemplates/delegate-terraform-example-module.ftl";
> 
>   private static final String onPremHelmRepoSuffix = "storage/harness-download/delegate-helm-chart/";
> ```

CAPITALIZE_CORRECTLY :slightly_smiling_face: 
___
*[`400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java` line 123 at r1](https://reviewable.io/reviews/harness/harness-core/51169#-NalMOzm8VzeXduu2pZ2:-NalMOzm8VzeXduu2pZ3:b-pnnb5o) ([raw file](https://github.com/harness/harness-core/blob/2f3a01eba252187b60a8dc7d3c7a25dffda37e32/400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java#L123)):*
> ```Java
>       return saasHelmRepo;
>     }
>     return null;
> ```

Don't return null, just throw exception (e.g. IllegalArgumentException)

**Comment by AnupamIO** [comment]
trigger uts

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java [hunk 1]
```diff
@@ -36,6 +36,9 @@ public class DelegateInstallationCommandServiceImpl implements DelegateInstallat
   private final MainConfiguration mainConfiguration;
   private static final String TERRAFORM_TEMPLATE_FLE = "/delegatetemplates/delegate-terraform-example-module.ftl";
 
+  private static final String ONPREM_HELM_REPO_SUFFIX = "storage/harness-download/delegate-helm-chart/";
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:39]: why not create a single variable in MainConfiguration, and set the url in manager config

**Review by xingchi-jin — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java (hunk 1 above)
Comment (AnupamIO) [reply 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:39]: adding saas url to config.yml, as onprem url is just a suffix so keeping it in this class only.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51169)* status: Some of the checks are too old. 3 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java` line 39 at r2](https://reviewable.io/reviews/harness/harness-core/51169#gh-1281191543:-Nc1bKCMAlCFS0yhbL-u:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4ef83fa06c3e03fe5a97d6db763e3948b647fc04/400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java#L39)):*
<details><summary><i>Previously, xingchi-jin (Xingchi Jin) wrote…</i></summary><blockquote>

why not create a single variable in MainConfiguration, and set the url in manager config 
</blockquote></details>

Done.

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger ut7

**Comment by AnupamIO** [comment]
trigger ut6

**Comment by AnupamIO** [comment]
trigger messagecheck

File: 360-cg-manager/config.yml [hunk 2]
```diff
@@ -762,6 +762,8 @@ disableDelegateMgmtInManager: false
 
 enableRedisForDelegateService: ${ENABLE_REDIS_DELEGATE_SERVICE:-false}
 
+saasDelegateHelmChartRepo: "https://app.harness.io/storage/harness-download/delegate-helm-chart/"
```
Comment (xingchi-jin) [inline 360-cg-manager/config.yml:765]: no very aligned with the implicit way of configuring smp url
can we just define a config called "delegateHelmChartRepo", and just set different values for saap and smp ?

Comment (AnupamIO) [reply 360-cg-manager/config.yml:765]: Xingchi, if we do that then that will mean we will have different set of configuration for smp and saas but at the time same time we have charter to make smp and saas configuration same. Adding different configurations will contradict that.

**Comment by AnupamIO** [comment]
trigger ut7

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger ut0

**Pull request merged**